### ISSUE TITLE
This analyzer does not detect mutating PC parameters via out and ref

### ISSUE THREAD

**Issue by DavidArno** [sailro/PrimaryConstructorAnalyzer#1]
Great analyzer, that almost does exactly what I was after. The only issue I have is that it doesn't detect all ways that the parameter could be mutated. Currently, it doesn't warn for mutation via `out` and `ref`, as per the example code below:
```cs
public sealed class C(int x)
{
    static void M1(out int a) => a = 0;

    static void M2(ref int a) => a = -1;

    public int M2()
    {
        M1(out x); // expecting PCA0001	Constructor parameter 'x' should not be mutated	here
        M2(ref x); // and here
        return x;
    }
}
```
Hopefully, you'll be able to fix that. 🤞

**Comment by sailro** [comment]
Thank you @DavidArno, I just added support for that, and pushed a new package version.

**Comment by DavidArno** [comment]
Wow, that's a fast turn-around. Thanks very much, @sailro.
